mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLOps.td

```diff
@@ -531,7 +523,7 @@ def SYCLConstructorOp : SYCL_Op<"constructor", [CallOpInterface]> {
 
   let arguments = (ins
     Arg<SYCLMemref, "The SYCL object being created", [MemWrite]>:$Dst,
-    Variadic<ConstructorArgs>:$Args,  
+    Arg<Variadic<AnyType>, "The constructor arguments", [MemRead]>:$Args,
```

Comment:
I've checked how memory effects are queried from operations and this should work, as an "`MemRead` position" would be generated for each variadic argument passed.